### PR TITLE
Run-10252-block-over-quota

### PR DESCRIPTION
**Pull request by jasonnovichRunAI**

### CODE REVIEW

File: docs/Researcher/scheduling/the-runai-scheduler.md [hunk 1]
```diff
@@ -64,6 +64,23 @@ Every new workload is associated with a Project. The Project contains a deserved
 * If the newly required resources, together with currently used resources, end up within the overall Project's quota and the requested node pool(s) quota, then the workload is ready to be scheduled as part of the guaranteed quota.
 * If the newly required resources together with currently used resources end up above the Project's quota or the requested node pool(s) quota, the workload will only be scheduled if there are 'spare' GPU resources within the same node pool but not part of this Project. There are nuances in this flow that are meant to ensure that a Project does not end up with an over-quota made entirely of interactive workloads. For additional details see below.
 
+### Blocking Over Quota
+
+Run:ai provides the ability to block over subscription of quotas by *Projects* or *Departments*.
+
+Over quota will be blocked under the following circumstances:
+
+* If a project’s quota request (with or without nodepools) increases the sum of all the projects quotas to more the department quota (for GPU and CPU, compute and memory).
```
Comment (natasharomm) [inline docs/Researcher/scheduling/the-runai-scheduler.md:73]:
```suggestion
* If a project’s quota request (with or without nodepools) increases the sum of all the department projects quotas to more than the department quota (for GPU and CPU, compute and memory).
```

File: docs/Researcher/scheduling/the-runai-scheduler.md [hunk 2]
```diff
@@ -64,6 +64,23 @@ Every new workload is associated with a Project. The Project contains a deserved
...
+* If a department’s quota decrease request (with or without nodepools) causes the sum of all projects quotas to surpass the department quota (for GPU and CPU, compute and memory).
+* If a project’s quota decrease request (with or without nodepools) causes the sum of all non-preemptible workloads to surpass that of the Project (for GPU and CPU, compute and memory).
```
Comment (natasharomm) [inline docs/Researcher/scheduling/the-runai-scheduler.md:75]: this point is only relevant for clusters with version 2.15+

Comment (natasharomm) [inline docs/Researcher/scheduling/the-runai-scheduler.md:75]:
```suggestion
* If a project’s quota decrease request (with or without nodepools) causes the sum of all allocated non-preemptible workloads in that project to surpass it (for GPU and CPU, compute and memory).
```

File: docs/Researcher/scheduling/the-runai-scheduler.md [hunk 3]
```diff
@@ -64,6 +64,23 @@ Every new workload is associated with a Project. The Project contains a deserved
 * If the newly required resources, together with currently used resources, end up within the overall Project's quota and the requested node pool(s) quota, then the workload is ready to be scheduled as part of the guaranteed quota.
 * If the newly required resources together with currently used resources end up above the Project's quota or the requested node pool(s) quota, the workload will only be scheduled if there are 'spare' GPU resources within the same node pool but not part of this Project. There are nuances in this flow that are meant to ensure that a Project does not end up with an over-quota made entirely of interactive workloads. For additional details see below.
 
+### Blocking Over Quota
+
+Run:ai provides the ability to block over subscription of quotas by *Projects* or *Departments*.
+
+Over quota will be blocked under the following circumstances:
```
[the remark this answers is not in the document]
Comment (jasonnovichRunAI) [reply docs/Researcher/scheduling/the-runai-scheduler.md:71]: OK, I understand the project/department won't get updated, but the sentence here needs a clarification as to why the update is not happening.
So maybe something like:

```suggestion
You project or department will not be updated due to quota over/under subscription issues caused by:
```

[the remark this answers is not in the document]
Comment (natasharomm) [reply docs/Researcher/scheduling/the-runai-scheduler.md:71]: yes this sounds good

File: docs/Researcher/scheduling/the-runai-scheduler.md [hunk 4]
```diff
@@ -64,6 +64,23 @@ Every new workload is associated with a Project. The Project contains a deserved
 * If the newly required resources, together with currently used resources, end up within the overall Project's quota and the requested node pool(s) quota, then the workload is ready to be scheduled as part of the guaranteed quota.
 * If the newly required resources together with currently used resources end up above the Project's quota or the requested node pool(s) quota, the workload will only be scheduled if there are 'spare' GPU resources within the same node pool but not part of this Project. There are nuances in this flow that are meant to ensure that a Project does not end up with an over-quota made entirely of interactive workloads. For additional details see below.
 
+### Blocking Over Quota
```
[the remark this answers is not in the document]
Comment (jasonnovichRunAI) [reply docs/Researcher/scheduling/the-runai-scheduler.md]:
```suggestion
### Limit Quota Over or Under Subscription
```

File: docs/Researcher/scheduling/the-runai-scheduler.md [hunk 5]
```diff
@@ -64,6 +64,23 @@ Every new workload is associated with a Project. The Project contains a deserved
...
+* If a project’s quota request (with or without nodepools) increases the sum of all the department projects quotas to more than the department quota (for GPU and CPU, compute and memory).
+* If a department’s quota decrease request (with or without nodepools) causes the sum of all projects quotas to surpass the department quota (for GPU and CPU, compute and memory).
+* If a project’s quota decrease request (with or without nodepools) causes the sum of all allocated non-preemptible workloads in that project to surpass it (for GPU and CPU, compute and memory).
+
+To configure blocking of over quota subscriptions:
+
+1. Press the *Tools and settings* icon, then go to *General*.
+2. Enable the *Limit quota over/under subscription* toggle.
+
+To disable the blocking of over quota subscription, disable the toggle.
```
Comment (jasonnovichRunAI) [inline docs/Researcher/scheduling/the-runai-scheduler.md:82]:
```suggestion
To disable the limiting of over quota subscription, disable the toggle.
```

File: docs/Researcher/scheduling/the-runai-scheduler.md [hunk 6]
```diff
@@ -64,6 +64,23 @@ Every new workload is associated with a Project. The Project contains a deserved
 * If the newly required resources, together with currently used resources, end up within the overall Project's quota and the requested node pool(s) quota, then the workload is ready to be scheduled as part of the guaranteed quota.
 * If the newly required resources together with currently used resources end up above the Project's quota or the requested node pool(s) quota, the workload will only be scheduled if there are 'spare' GPU resources within the same node pool but not part of this Project. There are nuances in this flow that are meant to ensure that a Project does not end up with an over-quota made entirely of interactive workloads. For additional details see below.
 
+### Quota Over or Under Subscription
+
+Run:ai provides the ability to block over subscription of quotas by *Projects* or *Departments*.
+
+Over quota will be blocked under the following circumstances:
```
Comment (jasonnovichRunAI) [inline docs/Researcher/scheduling/the-runai-scheduler.md:71]:
```suggestion
Over quota will be limited under the following circumstances:
```

File: docs/Researcher/scheduling/the-runai-scheduler.md [hunk 7]
```diff
@@ -64,6 +64,23 @@ Every new workload is associated with a Project. The Project contains a deserved
...
+
+* If a project’s quota request (with or without nodepools) increases the sum of all the department projects quotas to more than the department quota (for GPU and CPU, compute and memory).
+* If a department’s quota decrease request (with or without nodepools) causes the sum of all projects quotas to surpass the department quota (for GPU and CPU, compute and memory).
+* If a project’s quota decrease request (with or without nodepools) causes the sum of all allocated non-preemptible workloads in that project to surpass it (for GPU and CPU, compute and memory).
+
+To configure over or under quota subscriptions:
```
Comment (jasonnovichRunAI) [inline docs/Researcher/scheduling/the-runai-scheduler.md:77]:
```suggestion
To configure limiting over or under quota subscriptions:
```

**Pull request merged**